src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/INodeFilter.cs

```diff
@@ -53,5 +53,25 @@ public static string GetPropertyFilter(IEnumerable<INodeFilter> filters, Type ty
             }
             return string.Empty;
         }
+
+        /// <summary>
+        /// Appends a list of property filters to an URN query string
+        /// </summary>
+        public static string AddPropertyFilterToFilterString(string filterString, IEnumerable<INodeFilter> filters, Type type, ValidForFlag validForFlag)
+        {
+            if(String.IsNullOrEmpty(filterString))
+            {
+                return GetPropertyFilter(filters, type, validForFlag);
+            }
+            foreach(var value in filters)
```

Comment:
I think you do something like `foreach(var value in filters ?? Enumerable.Empty<INodeFilter>())` to be compliant with nullable compiler warnings for this statement.